src/plugins/soundChanger/index.tsx

```diff
@@ -0,0 +1,130 @@
+                                <Flex
+                                    key={sound}
+                                    flexDirection="row"
+                                    style={{ alignItems: "center", justifyContent: "space-between" }}
+                                >
+                                    <Text>{sound.slice(2, -4)}</Text>
+                                    <TextInput
+                                        className="sound-changer-link-input"
+                                        value={soundReplacements.find(r => r.name === sound)?.link ?? ""}
+                                        placeholder="Link to a sound..."
+                                        onChange={link => {
+                                            const index = soundReplacements.findIndex(r => r.name === sound);
+
+                                            if (index === -1) {
+                                                soundReplacements.push({ name: sound, link });
+                                            } else {
+                                                soundReplacements[index].link = link;
+                                            }
+
+                                            DataStore.set("SoundChange_replacements", soundReplacements);
+                                            update();
+                                        }}
+                                    />
+                                </Flex>
+                            )) : <Text>Enable the plugin to see available sounds!</Text>
+                        }
+                    </Flex>
+                </>
+            );
+        }
+    }
+});
+
+export default definePlugin({
+    name: "SoundChanger",
+    authors: [Devs.SpikeHD],
+    description: "Change Discord sounds to be your own!",
+
+    patches: [{
+        find: "./message1.mp3",
+        replacement: [
+            // This only runs once, making it a good source for dynamically retrieving all of the sounds the user can change
+            {
+                match: /var (.{1,2})=(\{.+?\});/g,
+                replace: "var $1=$2;$self.registerSoundFilenames($1);",
+            },
+            // This runs whenever Discord needs to play a sound
+            {
+                match: /(var .{1,2}=.{1,2}\((.{1,2})\));return (.{1,2}\(.{1,2}\))/g,
+                replace: "$1;return $self.getSound($2) ?? $3"
+            }
+        ]
+    }],
+
+    settings,
+
+    registerSoundFilenames: (names: Record<string, number>) => {
+        availableSounds = Vencord.Plugins.plugins.SoundChanger.settings?.store.showDetuneSounds ? Object.keys(names) : Object.keys(names).filter(name => !name.includes("detune"));
+    },
+
+    getSound: (name: string) => {
+        const replacement = soundReplacements.find(r => r.name === name);
+
+        if (replacement?.link) {
+            return replacement.link;
+        }
+
+        return null;
+    },
+
+    start: async () => {
```

Comment:
why not use proper methods? they're shorter
```suggestion
    async start() {
```